source/h2_frames.c

```diff
@@ -1227,7 +1227,9 @@ int aws_h2_encode_frame(
 }
 
 void aws_h2_frame_encoder_set_setting_header_table_size(struct aws_h2_frame_encoder *encoder, uint32_t data) {
-    aws_hpack_set_max_table_size(encoder->hpack, data);
+    /* Setting for dynamic table size changed from peer, we will udpate the dynamic table size when we encoder the next
+     * header block */
+    aws_hpack_set_max_table_size(encoder->hpack, data, true /*udpate table size*/);
     encoder->settings.header_table_size = data;
```

Comment:
seems like we can remove the `encoder->settings.header_table_size` variable, it serves no purpose